tests/unittests/helpers.py

```diff
@@ -428,37 +428,36 @@ def reRooted(self, root=None):
             self.patched_funcs.close()
 
 
-class CiRequestsMock(responses.RequestsMock):
-    def assert_call_count(self, url: str, count: int) -> bool:
-        """Focal and older have a version of responses which does
-        not carry this attribute. This can be removed when focal
-        is no longer supported.
-        """
-        if hasattr(super(), "_ensure_url_default_path"):
-            return super().assert_call_count(url, count)
-
-        def _ensure_url_default_path(url):
-            if isinstance(url, str):
-                url_parts = list(urlsplit(url))
-                if url_parts[2] == "":
-                    url_parts[2] = "/"
-                    url = urlunsplit(url_parts)
-            return url
-
-        call_count = len(
-            [
-                1
-                for call in self.calls
-                if call.request.url == _ensure_url_default_path(url)
-            ]
+def responses_assert_call_count(url: str, count: int) -> bool:
```

Comment:
Since Focal is in ESM now, we can drop this function definition and just use `responses.assert_call_count` at every call site instead.

Reply:
You can also cleanup t/unittests/sources/test_scaleway.py to avoid gating on         `if sys.version_info.minor >= 7:`